From 3b2b71bef46d3565db921c6a943c83ac0e8926b0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lu=C3=ADs=20Pereira?= <luis.artur.pereira@gmail.com>
Date: Mon, 2 Feb 2015 11:55:43 -0800
Subject: [PATCH 13/15] Avoid adding empty parent icon theme

"Icon Theme/Inherits" may assign an empty value, e.g. Flattr icon
theme

Port of commit
https://qt.gitorious.org/qt/qtbase/commits/9518b7e0f198c69d747d93eca5bb003944492c9f
---
 qiconfix/qiconloader.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/qiconfix/qiconloader.cpp b/qiconfix/qiconloader.cpp
index ce904db..193e2b8 100644
--- a/qiconfix/qiconloader.cpp
+++ b/qiconfix/qiconloader.cpp
@@ -243,6 +243,7 @@ QIconTheme::QIconTheme(const QString &themeName)
         // Parent themes provide fallbacks for missing icons
         m_parents = indexReader.value(
                 QLatin1String("Icon Theme/Inherits")).toStringList();
+        m_parents.removeAll(QString());
 
         // Ensure a default platform fallback for all themes
         if (m_parents.isEmpty()) {
-- 
1.9.0

